packages/core/src/utils/fragments.ts

```diff
@@ -0,0 +1,125 @@
+import type { Address, Hex } from "viem";
+
+import { FactoryCriteria } from "@/config/factories";
+import type { LogFilterCriteria } from "@/config/logFilters";
+
+/**
+ * Generates log filter fragments from a log filter.
+ *
+ * @param logFilter Log filter to be decompose into fragments.
+ * @returns A list of log filter fragments.
+ */
+export function buildLogFilterFragments({
+  address,
+  topics,
+  chainId,
+}: LogFilterCriteria & {
+  chainId: number;
+}) {
+  return buildFragments({
+    address,
+    topics,
+    chainId,
+    idCallback: (address_, topic0_, topic1_, topic2_, topic3_) =>
+      `${chainId}_${address_}_${topic0_}_${topic1_}_${topic2_}_${topic3_}`,
+  });
+}
+
+/**
+ * Generates factory fragments from a factory.
+ *
+ * @param factory Factory to be decomposed into fragments.
+ * @returns A list of factory fragments.
+ */
+export function buildFactoryFragments({
+  address,
```

Comment:
I find it _slightly_ misleading to put address and topics first like this, when here `address` means a static, singular factory contract address while in a log filter `address` means something different. Could be worth renaming either `FactoryCriteria.address` -> `FactoryCriteria.factoryAddress` or `FactoryCriteria.topics` -> `FactoryCriteria.childTopics`. Don't love either. Wdyt?